traitlets/traitlets.py

```diff
@@ -489,11 +488,24 @@ def _validate(self, obj, value):
         return value
 
     def _cross_validate(self, obj, value):
-        if hasattr(obj, '_%s_validate' % self.name):
-            cross_validate = getattr(obj, '_%s_validate' % self.name)
-            value = cross_validate(value, self)
+        try:
+            cb = getattr(obj, '_%s_validate' % self.name)
+            if cb in obj._trait_validators.get(self.name,[]):
+                raise
```

Comment:
This still seems awkward.  I don't think we should check for a conflict every time we try to validate something.  I think we should only check for a conflict on registration - that's a one-time cost, not a cost every time we validate.